tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -309,3 +274,64 @@ def parse_years(timestr):
     if len(years) == 2:
          years = list(range(years[0],years[1]+1))
     return years
+
+
+def queryRepos(inps=None):
+    """
+    Main workflow for querying supported GPS repositories for zenith delay information.
+    """
+
+    # Create specified output directory if it does not exist.
+    if not os.path.exists(inps.out):
+        os.mkdir(inps.out)
+
+    # Setup bounding box
+    if inps.bounding_box:
+        if isinstance([str(val) for val in inps.bounding_box.split()], list) and not os.path.isfile(inps.bounding_box):
+            try:
+                bbox = [float(val) for val in inps.bounding_box.split()]
+            except:
+                raise Exception(
+                    'Cannot understand the --bbox argument. String input is incorrect or path does not exist.')
+            # if necessary, convert negative longitudes to positive
+            if bbox[2] < 0:
+                bbox[2] += 360
+            if bbox[3] < 0:
+                bbox[3] += 360
+    # If bbox not specified, query stations across the entire globe
+    else:
+        bbox = [-90, 90, 0, 360]
+
+    # Handle station query
+    stats, origstatsFile = getStationList(
+        bbox=bbox, writeLoc=inps.out, userstatList=inps.station_file)
+
+    # iterate over years
+    for yr in inps.years:
+        statDF = downloadTropoDelays(
+            stats, yr, writeDir=inps.out, download=inps.download, verbose=inps.verbose)
+        statDF.to_csv(os.path.join(
+            inps.out, 'gnssStationList_overbbox_withpaths.csv'))
+
+    # Add lat/lon info
+    origstatsFile = pd.read_csv(origstatsFile)
+    statsFile = pd.read_csv(os.path.join(
+        inps.out, 'gnssStationList_overbbox_withpaths.csv'))
+    statsFile = pd.merge(left=statsFile, right=origstatsFile,
+                         how='left', left_on='ID', right_on='ID')
+    statsFile.to_csv(os.path.join(
+        inps.out, 'gnssStationList_overbbox_withpaths.csv'), index=False)
+    del statDF, origstatsFile, statsFile
+
+    # Extract delays for each station
+    getStationData(os.path.join(inps.out, 'gnssStationList_overbbox_withpaths.csv'),
+                   outDir=inps.out, returnTime=inps.returnTime)
+
+    if inps.verbose:
+        print('Completed processing')
+
+
+if __name__ == "__main__":
+    inps = cmdLineParse()
+
+    queryRepos(inps)
```

Comment:
I would unpack the arguments here, so that if this were to get called by another python program, that program would not need to somehow construct an object with attributes for the parameters. 
```suggestion
    queryRepos(
        inps.bounding_box,
        inps.verbose,
        ...
    )
```

Reply:
Agree with @Askaholic